setup_go.sh

```diff
@@ -0,0 +1,44 @@
+#!/bin/bash
+
+set -exo pipefail
```

Comment:
Is `-x` required? Usually we don't use it to avoid secrets leaks

Reply:
There should be no secret in this script and this might allow us to better debug go setup. If you still want me do remove it for consistency I can do it 👍